internal/app/service.go

```diff
@@ -484,6 +491,9 @@ func (svc *Service) Initialize() error {
 	svc.ctx.appCtx, svc.ctx.appCancelCtx = context.WithCancel(context.Background())
 	svc.ctx.appWg = &sync.WaitGroup{}
 
+	secretAddedSignal := make(chan struct{})
```

Comment:
Need a buffered channel so code pushing to the channel doesn't block. Linter will fail w/o this. Please make sure you run the linter (may need to install it).
```suggestion
	secretAddedSignal := make(chan struct{}, 1)
```